### PR TITLE
66 handel search page

### PR DESCRIPTION
**Pull request by husamkamal**
in the last commit : 
  - setLoading inside getCars function
  - add try and catch when fetching data to handle errors 
  - =defaultValue={state?.brand || null} in brands
  - CustomSkeleton() remove it to components folder 
  -

### CODE REVIEW

**Review by tariqabusada — APPROVED** [review]

**Review by abdou059 — APPROVED** [review]

File: client/src/components/CarCard/index.tsx [hunk 1]
```diff
@@ -20,7 +20,7 @@ export default function CarCard({
   carName,
   quality,
   price,
-  mailage,
+  milage,
```
Comment (mai-mohammed) [inline client/src/components/CarCard/index.tsx]: mileage

**Review by mai-mohammed — APPROVED** [review]

File: client/src/components/CarsFilter/index.tsx [hunk 2]
```diff
@@ -44,44 +46,93 @@ const getYears = ():Array<string> => {
...
   };
   const changeMaxPrice = (event: React.ChangeEvent<HTMLInputElement>) => {
     const num = Number(event.target.value);
     setMaxPrice(num);
   };
 
   useEffect(() => {
-    // here we will fetch the data and set it on a props that it defined in parent component
-  }, [brand, model, mileage, year, fuel, isGoodPrice]);
+    setLoading(true);
```
Comment (LinaYahya) [inline client/src/components/CarsFilter/index.tsx]: setLoading to true inside getCars, As I gurantee that loading set to true when request is about to send

File: client/src/components/CarsFilter/index.tsx [hunk 3]
```diff
@@ -44,44 +46,93 @@ const getYears = ():Array<string> => {
...
+    }
+    if (isGoodPrice) {
+      params.goodPrice = 1;
+    }
+    if (mileage !== 0) {
+      params.mileage = mileage;
+    }
+    const getCars = async () => {
+      // eslint-disable-next-line max-len
+      const response: CarsData = await httpInstance.get('/cars?', { params });
```
Comment (LinaYahya) [inline client/src/components/CarsFilter/index.tsx]: What if there's an error!
create try and catch block and create an error state if there's an error, maybe we can use a snackbar component to tell the user that there's unexpected error

File: client/src/components/CarsFilter/index.tsx [hunk 4]
```diff
@@ -44,44 +46,93 @@ const getYears = ():Array<string> => {
...
+      setCars(response.data.rows);
+      setPagination(response.data.count);
+      setLoading(false);
+    };
+    getCars();
+  }, [brand, model, mileage, year, fuel, isGoodPrice, pageNumber, search, maxPrice]);
 
   return (
     <section className="filter">
```
Comment (LinaYahya) [inline client/src/components/CarsFilter/index.tsx:157]: change the default value for brand to `=defaultValue={state?.brand || null}` as the autocomplete doesn't expect to receive an empty string as a default value

File: client/src/helpers/skeleton.tsx [hunk 5]
```diff
@@ -0,0 +1,23 @@
+import Skeleton from '@mui/material/Skeleton';
+
+export default function CustomSkeleton() {
+  return (
+    <div
```
Comment (LinaYahya) [inline client/src/helpers/skeleton.tsx]: move this to the components folder,  any try to change the style to be more closer to the car card component

File: client/src/pages/Cars/index.tsx [hunk 6]
```diff
@@ -1,8 +1,6 @@
-import CarsFilter from '../../components/CarsFilter';
-
 function Cars() {
   return (
-    <CarsFilter />
+    <h1>sd</h1>
```
Comment (LinaYahya) [inline client/src/pages/Cars/index.tsx]: this folder should be the search page, SO use one of them either this or the search!

File: client/src/pages/Search/index.tsx [hunk 7]
```diff
@@ -1,52 +1,84 @@
...
+          )
+
+          : (cars?.length ? (
+            <div className="car_wrapper">
+              {cars.map((e) => (
+                <CarCard
+                  key={e.id}
+                  image={
+                  // eslint-disable-next-line max-len
+                  e.images[0]?.image || 'https://thumbs.dreamstime.com/b/no-image-available-icon-photo-camera-flat-vector-illustration-132483141.jpg'
```
Comment (LinaYahya) [inline client/src/pages/Search/index.tsx]: notice that the car width for the components doesn't have images is not the same for others, So try to fix style to give all cards the same width so they will be aligned together at the container

File: client/src/pages/Search/index.tsx [hunk 8]
```diff
@@ -1,52 +1,84 @@
...
+                  isGoodPrice={e.isGoodPrice}
+                />
+              ))}
+              <Pagination
+                sx={{
+                  margin: '2rem 0rem',
+                  marginLeft: '10rem',
+                }}
+                onChange={paginationHandler}
+                count={pagination / 9}
```
Comment (LinaYahya) [inline client/src/pages/Search/index.tsx]: use  `ceil` here

File: client/src/pages/Search/index.tsx [hunk 9]
```diff
@@ -1,52 +1,84 @@
...
+                variant="outlined"
+              />
+
+            </div>
+          ) : <h2 style={{ fontSize: '2rem' }}>NO results found</h2>)}
```
Comment (LinaYahya) [inline client/src/pages/Search/index.tsx]: create a custom component for no results

File: client/src/pages/Search/index.tsx [hunk 10]
```diff
@@ -1,52 +1,84 @@
...
-  description:
-  `Posuere consectetur est at lob ortis. Aenean eu l eo quam Pellentesque
-   ornare sem lacini ... more`,
-  isGoodPrice: true,
-
-}];
 function Search() {
+  const [cars, setCars] = useState<CarsRow | []>([]);
+  const [pagination, setPagination] = useState <CarsCount>(1);
+  const [pageNumber, setPageIndex] = useState<number>(1);
```
Comment (LinaYahya) [inline client/src/pages/Search/index.tsx]:
```suggestion
  const [currentPage, setCurrentPAge] = useState<number>(1);
```
use meaningful, descriptive names

**Review by LinaYahya — CHANGES_REQUESTED** [review]

**Review by LinaYahya — APPROVED** [review]

**Pull request merged**